regression/clear_datastore.py

```diff
@@ -46,37 +50,39 @@ def fetch_keys(dataset, kind, fetch_max=FETCH_MAX, query=None, cursor=None):
 
 def get_ancestors(entities):
     # NOTE: A key will always have at least one path element.
-    key_roots = [entity.key().path[0] for entity in entities]
-    # Turn into hashable type so we can use set to get unique roots.
-    # Also sorted the items() to ensure uniqueness.
-    key_roots = [tuple(sorted(root.items())) for root in key_roots]
-    # Cast back to dictionary.
-    return [dict(root) for root in set(key_roots)]
+    key_roots = [entity.key.flat_path[:2] for entity in entities]
```

Comment:
No because the parent might have more than 1 element in the path.

For example:
```python
>>> from gcloud import datastore
>>> from gcloud.datastore.key import Key
>>>
>>> datastore.set_default_dataset_id('foo')
>>>
>>> k = Key('Foo', 1, 'Bar', 2, 'Baz', 3)
>>> k
<Key[{'kind': 'Foo', 'id': 1}, {'kind': 'Bar', 'id': 2}, {'kind': 'Baz', 'id': 3}], dataset=foo>
>>> k.parent
<Key[{'kind': 'Foo', 'id': 1}, {'kind': 'Bar', 'id': 2}], dataset=foo>
```